### PR TITLE
[SYCL][ESIMD] Test half type conversion under ESIMD emulator

### PR DESCRIPTION
**Pull request by fineg74**
Complementary compiler PR: https://github.com/intel/llvm/pull/6592

### CODE REVIEW

File: SYCL/ESIMD/api/slm_gather_scatter_heavy.cpp [hunk 1]
```diff
@@ -7,9 +7,7 @@
 //===----------------------------------------------------------------------===//
 // REQUIRES: gpu
 // UNSUPPORTED: cuda || hip
-// TODO: esimd_emulator fails due to outdated memory intrinsic
-// XFAIL: esimd_emulator
-// RUN: %clangxx -fsycl %s -o %t.out
+``````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````` // RUN: %clangxx -fsycl %s -o %t.out
```
Comment (dongkyunahn-intel) [inline SYCL/ESIMD/api/slm_gather_scatter_heavy.cpp]: Typo?

Comment (fineg74) [reply SYCL/ESIMD/api/slm_gather_scatter_heavy.cpp]: Fixed

**Review by dongkyunahn-intel — DISMISSED** [review]

**Comment by dongkyunahn-intel** [comment]
@fineg74 , one more test (ext_math.cpp) can be enabled and one test (svm_gather_scatter.cpp) needs to be changed for half-type support. Can I add a commit for the fix in this PR?

**Comment by fineg74** [comment]
> @fineg74 , one more test (ext_math.cpp) can be enabled and one test (svm_gather_scatter.cpp) needs to be changed for half-type support. Can I add a commit for the fix in this PR?

I see it is enabled for emulator already in my branch.

File: SYCL/ESIMD/regression/half_conversion_test.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  Ty VAL = (Ty)inc;
+
+  try {
+    buffer<Ty, 1> buf(data, range<1>(1));
+    q.submit([&](handler &cgh) {
+      std::cout << "Running on "
+                << q.get_device().get_info<::sycl::info::device::name>()
+                << "\n";
+      auto acc = buf.template get_access<access::mode::read_write>(cgh);
+      cgh.single_task<test_id<Ty>>([=]() SYCL_ESIMD_KERNEL {
```
Comment (v-klochkov) [inline SYCL/ESIMD/regression/half_conversion_test.cpp]: non-blocking comment. This should work properly without explicit name set:
```suggestion
      cgh.single_task([=]() SYCL_ESIMD_KERNEL {
```

File: SYCL/ESIMD/regression/half_conversion_test.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+//==- half_conversion_test.cpp - Test for half conversion under ESIMD_EMULATOR
+// backend -==/
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
```
Comment (v-klochkov) [inline SYCL/ESIMD/regression/half_conversion_test.cpp]: Please don't use CL. There is an ongoing activity to replace CL with sycl. See: https://github.com/intel/llvm-test-suite/pull/1078

**Review by v-klochkov — COMMENTED** [review]
Looks good, I have 2 comments.

**Review by v-klochkov — APPROVED** [review]

**Comment by pvchupin** [comment]
@kbobrovs, please always merge test changes together with compiler changes. These tests fail now in pre-commit PRs.

**Review by pvchupin — APPROVED** [review]

**Comment by kbobrovs** [comment]
> @kbobrovs, please always merge test changes together with compiler changes. These tests fail now in pre-commit PRs.

right, sorry - this time took a while.

**Pull request merged**